src/server-extension/resolvers/ChannelsResolver/index.ts

```diff
@@ -76,6 +79,49 @@ export class ChannelsResolver {
     return result
   }
 
+  @Query(() => [TopSellingChannelsResult])
+  async topSellingChannels(
+    @Args() args: TopSellingChannelsArgs,
+    @Info() info: GraphQLResolveInfo,
+    @Ctx() ctx: Context
+  ): Promise<TopSellingChannelsResult[]> {
+    const currentTime = new Date().getTime()
+    const cacheKey = `top-selling-channels-${args?.periodDays}`
+    const cachedData = cache?.get(cacheKey)
+    const dataTTL = cache?.getTtl(cacheKey)
+
+    const listQuery = buildTopSellingChannelsQuery(args, info, ctx)
+
+    // check if we need to update the cache
+    console.log('cache data', cachedData, dataTTL)
+    if (!cachedData || !dataTTL || dataTTL - currentTime < 5 * 1000) {
+      if (cachedData) {
+        console.log('cache expires soon, updating ', cacheKey)
+        // data is in the cache, so just trigger the update
+        setTimeout(() => {
+          ctx.openreader
+            .executeQuery(listQuery)
+            .then((result) => {
+              console.log('Result', result)
+              cache?.set(cacheKey, result)
+            })
+            .catch(() => {
+              // do nothing
+            })
+        }, 0)
+        return cachedData as TopSellingChannelsResult[]
+      } else {
+        console.log('cache miss', cacheKey)
+        // there is no data in cache, so we need to wait for it before returning it
+        const result = await ctx.openreader.executeQuery(listQuery)
+        cache?.set(cacheKey, result)
```

Comment:
The main issue with the cache you've implemented it that it's indifferent to the query parameters provided by the user. For example, if the user includes `where: { channel: { title_contains: "goku" } }` parameter in the query, only the channels containing `goku` in the title will be cached. Then if the same, or a different user executes another query without any `where` parameters provided, they will still see only those channels, which were cached based on the previous parameters (ie. only the channels containing `goku` in the title).